tests/test_kpcovr.py

```diff
@@ -0,0 +1,176 @@
+        and that the prediction error increases with `mixing`
+        """
+        prev_error = -1.0
+
+        for i, mixing in enumerate(np.linspace(0, 1, 11)):
+
+            pcovr = KPCovR(mixing=mixing, n_components=2, tol=1e-12)
+            pcovr.fit(self.X, self.Y)
+
+            _, error = pcovr.score(self.X, self.Y)
+
+            with self.subTest(error=error):
+                self.assertFalse(np.isnan(error))
+            with self.subTest(error=error, alpha=round(mixing, 4)):
+                self.assertGreaterEqual(error, prev_error - self.error_tol)
+
+            prev_error = error
+
+    def test_reconstruction_errors(self):
+        """
+        This test checks that KPCovR returns a non-null reconstructed X
+        and that the reconstruction error decreases with `mixing`
+        """
+
+        prev_error = 1.0
+
+        for i, mixing in enumerate(np.linspace(0, 1, 11)):
+            pcovr = KPCovR(mixing=mixing, n_components=2, tol=1e-12)
+            pcovr.fit(self.X, self.Y)
+
+            error, _ = pcovr.score(self.X, self.Y)
+
+            with self.subTest(error=error):
+                self.assertFalse(np.isnan(error))
+            with self.subTest(error=error, alpha=round(mixing, 4)):
+                self.assertLessEqual(error, prev_error + self.error_tol)
+
+            prev_error = error
+
+
+class KPCovRInfrastructureTest(KPCovRBaseTest):
+    def test_nonfitted_failure(self):
+        """
+        This test checks that KPCovR will raise a `NonFittedError` if
+        `transform` is called before the model is fitted
+        """
+        model = KPCovR(mixing=0.5, n_components=2, tol=1e-12)
+        with self.assertRaises(exceptions.NotFittedError):
+            _ = model.transform(self.X)
+
+    def test_no_arg_predict(self):
+        """
+        This test checks that KPCovR will raise a `ValueError` if
+        `predict` is called without arguments
+        """
+        model = KPCovR(mixing=0.5, n_components=2, tol=1e-12)
+        model.fit(self.X, self.Y)
+        with self.assertRaises(ValueError):
+            _ = model.predict()
+
+    def test_T_shape(self):
+        """
+        This test checks that KPCovR returns a latent space projection
+        consistent with the shape of the input matrix
+        """
+        n_components = 5
+        pcovr = KPCovR(mixing=0.5, n_components=n_components, tol=1e-12)
+        pcovr.fit(self.X, self.Y)
+        T = pcovr.transform(self.X)
+        self.assertTrue(check_X_y(self.X, T, multi_output=True))
+        self.assertTrue(T.shape[-1] == n_components)
+
+
+class KernelTests(KPCovRBaseTest):
+    def test_kernel_types(self):
+        """
+        This test checks that KPCovR can handle all kernels passable to
+        sklearn kernel classes
+        """
+        kernel_params = {
+            "poly": {"degree": 2},
+            "rbf": {"gamma": 3.0},
+            "sigmoid": {"gamma": 3.0, "coef0": 0.5},
+        }
```

Comment:
Should we also test for precomputed kernels and kernels constructed from callable functions? (`KernelRidge` also accepts these as kernel options, for example -- from a quick glance the KPCovR class might be able to handle it as well)